roles/rsyslog/tasks/main.yaml

```diff
@@ -1,43 +1,31 @@
----
 - name: Install/Update required packages
   package:
-    name: "{{ lookup('flattened', rsyslog_base_packages, rsyslog_tls_packages if rsyslog_pki else [], rsyslog_packages) }}"
+    name: "{{ (__rsyslog_base_packages) + (__rsyslog_tls_packages if rsyslog_pki else []) + (rsyslog_additional_packages) }}"
     state: latest
   when:
     - rsyslog_enabled | bool
-    - not use_rsyslog_image | default(false) | bool
-  ignore_errors: yes
-  tags:
-    - skip_ansible_lint
+
+- name: Gather package facts
+  package_facts:
+    manager: auto
 
 - name: Get rsyslog version
-  package:
-    list: rsyslog
-  register: rsyslog_version
+  set_fact:
+    __rsyslog_version: "{{ ansible_facts.packages['rsyslog'][0].version }}"
```

Comment:
Hmm I thought about a problem, what if rsyslog is not installed here? With @vcrhonek we have been discussing a similar problem in the timesync role.

Reply:
> Hmm I thought about a problem, what if rsyslog is not installed here? 

Could it happen even if rsyslog is installed with the package module just above this task?

Reply:
yes, think about check mode.

Reply:
> yes, think about check mode.

I see.  Do you recommend to add `check_mode: no` to the install packages tasks?

Reply:
No, this would defeat the purpose of using check mode, which is to not make changes to the system.

Reply:
Indeed...  We can do either 1) quit with no such package error here or 2) pretend having an rsyslog package and keep running since it is a dry-run.  Which would be better?

Reply:
I would say some version of 2) since I don't think we should fail in check mode if we would not fail in normal mode. We could even skip the rest of the role since we properly reported changed status.

Reply:
Here's a fix for a similar problem in timesync: https://github.com/linux-system-roles/timesync/pull/60

Reply:
Thank you so much, @pcahyna.  I tested this diff and verified it works as expected.
https://github.com/nhosoi/logging/compare/best_practices...nhosoi:package_version

Once pr/89 is done, I'm opening a pr with it.
https://github.com/linux-system-roles/logging/pull/89

Reply:
@nhosoi I did not intend to say that you must follow the patch taken in timesync. Rather we should determine which approach is better and use it consistently.
(yet another approach would be to not support check mode on first run of the role until Ansible 2.7 is deprecated and we can solve the problem much more elegantly using the default filter).